### PR TITLE
[WIP] FEA Implementation of a wrapper for bootstraping any sklearn.metrics-like

### PR DESCRIPTION
**Pull request by vitaliset**
First version of a wrapper that bootstrap your sample and compute metrics on top of the samples.

Using it is as simple as:
```python
f4_score = functools.partial(fbeta_score, beta=4, average='macro')
metric_list = bootstrap_metric(
    metric=f4_score,
    y_true=y_test,
    y_pred=probs,
)
```

I would like some comments on the design, since, as far as I know, sklearn doesn't have any function that proposes to do something similar (modify the behavior of a metric so that it does more things).

TODO:
- [ ] Docstring.
- [ ] More tests (I would love suggestions).
- [ ] Others TODO in code.

____

Next steps (after merge):
- TBD: Create specifications that use this function... For instance mean and std (?) or ‘percentile’, ‘basic’, ‘bca’ (from [scipy.stats.bootstrap](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.bootstrap.html)).
- TBD: Create a "DisplayBootstrapEvaluation" class (histogram of the metrics obtained).

### CODE REVIEW

**Review by dihanster — COMMENTED** [review]
Great job overall, Carlo. This method will be very useful!

I just left a few comments, which we can improve!

**Review by vitaliset — COMMENTED** [review]
Resolving W391 linting error.

**Review by jeancmaia — COMMENTED** [review]
Carlo. After reviewing your code and discussions, I came across one idea. It regards our library's whole possible Bootstrap suite and how we perceive this base method, `bootstrap_metric`.

Many great evaluative tools, such as DisplayBootstrapEvaluation and ConfidenceIntervalEvaluation, would be built on top of the same base method. This backbone would be a method that bootstraps n times and returns an array with the same size, holding the results of each sample. Therefore, gazing at this PR and mid-term horizon, it wouldn't be better for this method to return the bootstrap list, as opposed to mean and std.
